ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -40,97 +40,171 @@ const styles = reactStyles({
+    <div style={styles.formGrid}>
+      <label htmlFor='num-workers' style={{marginRight: '.25rem'}}>Workers</label>
+      <InputNumber id='num-workers'
+        showButtons
+        decrementButtonClassName='p-button-secondary'
+        incrementButtonClassName='p-button-secondary'
+        value={updatedNumWorkers}
+        inputStyle={{padding: '.75rem .5rem', width: '2rem'}}
+        onChange={({value}) => {
+          setUpdatedNumWorkers(value);
+          if (updatedNumWorkers < updatedPreemtible) {
+            setUpdatedPreemptible(updatedNumWorkers);
+          }
+        }}
+        min={2}/>
+      <label htmlFor='num-preemptible' style={{marginRight: '.25rem'}}>Preemptible</label>
+      <InputNumber id='num-preemptible'
+        showButtons
+        decrementButtonClassName='p-button-secondary'
+        incrementButtonClassName='p-button-secondary'
+        value={updatedNumWorkers < updatedPreemtible ? updatedNumWorkers : updatedPreemtible}
+        inputStyle={{padding: '.75rem .5rem', width: '2rem'}}
+        onChange={({value}) => setUpdatedPreemptible(value)}
+        min={0}
+        max={updatedNumWorkers}/>
+      <div style={{gridColumnEnd: 'span 2'}}/>
+      <MachineSelector machineType={workerMachineType} onChange={setUpdatedWorkerMachine} updatedMachine={updatedWorkerMachine}/>
+      <DiskSizeSelection diskSize={workerDiskSize} onChange={setUpdatedDiskSize} updatedDiskSize={updatedDiskSize} />
+    </div>
+  </fieldset>;
 };
 
 export const RuntimePanel = withCurrentWorkspace()(({workspace}) => {
-  const [updatedDiskSize, setUpdatedDiskSize] = useState(null);
-  const [updatedMachine, setUpdatedMachine] = useState(null);
   const runtimeOps = useStore(runtimeOpsStore);
   const [currentRuntime, setRequestedRuntime] = useCustomRuntime(workspace.namespace);
 
   const activeRuntimeOp: RuntimeOperation = runtimeOps.opsByWorkspaceNamespace[workspace.namespace];
   const {status = RuntimeStatus.Unknown, toolDockerImage = '', dataprocConfig = null, gceConfig = {}} = currentRuntime || {};
   const masterMachineType = !!dataprocConfig ? dataprocConfig.masterMachineType : gceConfig.machineType;
   const masterDiskSize = !!dataprocConfig ? dataprocConfig.masterDiskSize : gceConfig.bootDiskSize;
-  const updatedMachineType = updatedMachine && updatedMachine.name;
 
-  const isDataproc = (currentRuntime && !!currentRuntime.dataprocConfig);
-  const runtimeChanged = updatedMachine || updatedDiskSize;
+  const [updatedDiskSize, setUpdatedDiskSize] = useState(masterDiskSize);
+  const [updatedMachine, setUpdatedMachine] = useState(null);
+  const [updatedCompute, setUpdatedCompute] = useState<ComputeType>(dataprocConfig ? ComputeType.dataproc : ComputeType.standard);
+  const [updatedDataprocConfig, setUpdatedDataprocConfig] = useState();
+
+  const updatedMachineType = updatedMachine && updatedMachine.name;
+  const runtimeChanged = updatedMachine || updatedDiskSize !== masterDiskSize || updatedDataprocConfig;
```

Comment:
Outside of the scope of this ticket, but it would be nice to clean up how we track whether gce config has changed so that it can work the same as what you're doing... on Monday I'll take a look at what tickets we have for the epic and what else we need